[FR-113] NextJS Message Action Buttons

tylergeorges opened the pull request:
This PR adds message actions such as:

- Speak Message
- Copy Message ID
- Copy Message Link

(Delete message action will be implemented in separate PR)

Resolves FR-113

advaith1 reviewed (changes requested):
* buttons should use cursor: pointer
* buttons should move down on click like in Discord
* spoken text includes `message.speak.said` instead of the correct string

first two might be message renderer issues

tylergeorges commented:
@advaith1 
- for cursor pointer i fixed it by setting cursor: pointer as default for buttons

- message buttons moving down on click is a message renderer thing

- fixed the speak message not resolving localized string

JohnyTheCarrot commented on src/components/Providers/MessageRenderer.tsx:
```diff
@@ -133,6 +140,52 @@ export const MessageRenderer = ({ children }: MessageRendererWrapperProps) => {
+
+  const speakMessage = (message: APIMessage) => {
+    const action = message.referenced_message
+      ? `${t('message.speak.repliedto')} ${message.referenced_message.author.username}`
```
This isn't the way. The location of the username in the string should be localizable too, different languages put them in different places. Example: "{x} replied to {y}" -> "{x} heeft op {y} geantwoord" (not the same string in the example)
